pkg/kubelet/cm/dra/manager.go

```diff
@@ -126,39 +114,28 @@ func (m *ManagerImpl) PrepareResources(pod *v1.Pod) error {
 
 				klog.V(3).InfoS("NodePrepareResource succeeded", "response", response)
 
-				// NOTE: Passing CDI device names as annotations is a temporary solution
-				// It will be removed after all runtimes are updated
-				// to get CDI device names from the ContainerConfig.CDIDevices field
-				annotations, err := generateCDIAnnotations(resourceClaim.UID, driverName, response.CdiDevices)
-				if err != nil {
-					return fmt.Errorf("failed to generate container annotations, err: %+v", err)
-				}
-
-				// Cache prepared resource
-				err = m.cache.add(
+				// TODO: We are adding the claimInfo struct to the cache and syncing it to the checkpoint *after* the NodePrepareResource
+				// call has completed. This will cause issues if the kubelet gets restarted between NodePrepareResource and syncToCheckpoint.
+				// It will result in not calling NodeUnprepareResource for this claim because not claimInfo will be synced back to the cache
+				// for it after the restart. We need to resolve this issue before moving to beta.
+				claimInfo, err := newClaimInfo(
+					driverName,
+					resourceClaim.UID,
 					resourceClaim.Name,
 					resourceClaim.Namespace,
-					&claimInfo{
-						driverName:  driverName,
-						claimUID:    resourceClaim.UID,
-						claimName:   resourceClaim.Name,
-						namespace:   resourceClaim.Namespace,
-						podUIDs:     sets.New(string(pod.UID)),
-						cdiDevices:  response.CdiDevices,
-						annotations: annotations,
-					})
+					sets.New(string(pod.UID)),
+					response.CdiDevices)
 				if err != nil {
-					return fmt.Errorf(
-						"failed to cache prepared resource, claim: %s(%s), err: %+v",
-						resourceClaim.Name,
-						resourceClaim.UID,
-						err,
-					)
+					return fmt.Errorf("newClaimInfo failed, claim UID: %s, claim name: %s, claim namespace: %s, err: %+v",
+						resourceClaim.UID, resourceClaim.Name, resourceClaim.Namespace, err)
 				}
+
+				m.cache.add(claimInfo)
+				m.cache.syncToCheckpoint()
```

Comment:
Do we actually sync to checkpoint? I'd say we create a checkpoint, no?

Reply:
It is overwriting the same file over and over again -- hence syncing the state from what's cached in memory back to the file.

Reply:
ok. I thought that `writeToCheckpoint` or `storeToCheckpoint` or something similar would be better, but I don't feel strongly about it.

Reply:
Let's keep it as `sync` if it really causes confustion down the road we can always change it -- it is an internal API.